native/src/main/java/io/ballerina/stdlib/grpc/Message.java

```diff
@@ -177,6 +205,59 @@ public Message(
         } else if (type.getTag() == TypeTags.ARRAY_TAG) { // for array values inside structs
             bArray = ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_ANYDATA));
             bMessage = bArray;
+        } else if (GOOGLE_PROTOBUF_ANY_MESSAGE_NAME.equals(messageName) && input != null) {
```

Comment:
```suggestion
        } else if (isAnyTypedMessage && input != null) {
```
i think we can use `isAnyTypedMessage` here

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/509/commits/3b17f47c72f59298dfa72cc8ac04c920f5e67db0